core/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -179,7 +179,7 @@ public GroupShardsIterator allActiveShardsGrouped(String[] indices, boolean incl
                     if (shardRouting.active()) {
                         set.add(shardRouting.shardsIt());
                         if (includeRelocationTargets && shardRouting.relocating()) {
-                            set.add(new PlainShardIterator(shardRouting.shardId(), ImmutableList.of(shardRouting.targetRoutingIfRelocating())));
+                            set.add(new PlainShardIterator(shardRouting.shardId(), ImmutableList.of(shardRouting.buildTargetRelocatingShard())));
```

Comment:
You can't grow an Arrays.asList instance, but you can still use `set(int index, T value)`